_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,265 @@
+
+My team is currently working on the development of [Allegro Ads](https://allegro.pl/ads), which is an advertising tool for our ecommerce platform sellers. It is a whole panel for viewing statistics, displaying advertising campaigns. We provide sponsored offers for our allegro site, facebook, google and recently even for on the partner websites. Our sellers pay us for the opportunity to advertise their products, and at the same time we strive to ensure that these ads accurately reach people potentially interested in buying, thereby generating profits for sellers.
+
+The main technology stack is standard and still pretty cool. The core of the project is written in React, and significant parts have been rewritten to Typescript. From the beginning, for testing purposes, we use Jest. In my opinion this is a very good and stable tool, easy to work with. It has quite a low entry threshold, even if someone hasn’t had too much experience with tests before. The main advantages of Jest for me are minimal configuration and quite good documentation. But for testing react components we currently have two libraries and I'll explain why in a moment.
+
+The 2.0. version of platform Allegro Ads was released about 6 years ago, with an enzyme library on a board. A year ago the team added a react testing library to package.json. We have both packages so far. **Recently we officially decided to slowly rewrite the enzyme library for good.** We added the rule to our docc (Declaration of Code Convention) that the new components are tested just in rtl, the old ones will be rewritten when existing parts of the code will be touched.
+
+## Migration effects so far
+
+Currently we have got **592 test files** (counting after the file names with “spec” phrase). In these files we can also find **709 of unique “describe”** usage for wrapping a series of tests, and **1796 unique test cases**. Half of tests are written only in jest, but a significant amount of tests for better testing use extra libraries. 
+
+React testing library is used in **131 files**, which is around **22% of all test files**. 
+Enzyme tests are in **137 files** which is around **23%**.  
+
+So the enzyme is still used more often. However, keep in mind that we have been using Enzyme in this project from the beginning and rtl for a year. So I'm quite stunned by these numbers. I did not think that the use of rtl would increase so significantly in a year. Of course, there is still a long way to completely eliminate Enzyme, especially among other project priorities, but now I'm convinced that small steps will be enough to do that.
+
+## Summary
+
+Hopefully this article showed you that rewriting a library from one to the other doesn't have to be difficult. As programmers, we will encounter such situations frequently. I think that the effort put into replacing old libraries with newer ones and keeping your code free of technological debt will bring tangible results for developers, project and, finally, users.
```

Comment:
we are not rewriting library :)

Reply:
I've change this line, maybe it was a bad word. Do you think that I should change also the title? I assume that "Why we should rewrite" sounds good and shows that article is about changing libraries. Not about rewriting one. ;)